sources/corepackUtils.ts

```diff
@@ -1,20 +1,20 @@
-import {FileHandle}                                            from 'fs/promises';
-import fs                                                      from 'fs';
-import type {Dir}                                              from 'fs';
-import Module                                                  from 'module';
-import path                                                    from 'path';
-import semver                                                  from 'semver';
-
-import * as engine                                             from './Engine';
-import * as debugUtils                                         from './debugUtils';
-import * as folderUtils                                        from './folderUtils';
-import * as fsUtils                                            from './fsUtils';
-import * as httpUtils                                          from './httpUtils';
-import * as nodeUtils                                          from './nodeUtils';
-import * as npmRegistryUtils                                   from './npmRegistryUtils';
-import {RegistrySpec, Descriptor, Locator, PackageManagerSpec} from './types';
+import {createHash}                                                                           from 'crypto';
+import {once}                                                                                 from 'events';
+import {FileHandle}                                                                           from 'fs/promises';
+import fs                                                                                     from 'fs';
+import type {Dir}                                                                             from 'fs';
+import Module                                                                                 from 'module';
+import path                                                                                   from 'path';
+import semver                                                                                 from 'semver';
+
+import * as engine                                                                            from './Engine';
+import * as debugUtils                                                                        from './debugUtils';
+import * as folderUtils                                                                       from './folderUtils';
+import * as fsUtils                                                                           from './fsUtils';
+import * as httpUtils                                                                         from './httpUtils';
+import * as nodeUtils                                                                         from './nodeUtils';
+import * as npmRegistryUtils                                                                  from './npmRegistryUtils';
+import {RegistrySpec, Descriptor, Locator, PackageManagerSpec, BinList, BinSpec, InstallSpec} from './types';
```

Comment:
Consider splitting it in two exports to avoid needing vertical scroll to see the imports
```suggestion
import {RegistrySpec, Descriptor, Locator, PackageManagerSpec} from './types';
import {BinList, BinSpec, InstallSpec} from './types';
```